src/main/java/agh/ics/oop/AbstractWorldMap.java

```diff
@@ -0,0 +1,35 @@
+package agh.ics.oop;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public abstract class AbstractWorldMap {
+    //wydaje mi sie, ze przeniesienie tutaj listy animals nie ma sensu "ideowo", bo wiekszy sens "ideowo" mialoby dziedziczenie GrassField po RectangularMap
+    //aczkolwiek przenioslem ja tutaj, aby spelnic polecenie "Dodaj klasę abstrakcyjną AbstractWorldMap, która zawiera kod wspólny dla tych klas."
```

Comment:
miałby tu 100% racji gdyby nie fakt, że jednak te klasy trochę się od siebie różnią: mapa prostokątna ma ograniczony rozmiar, a więc posiada atrybuty `width` i `height`. Gdyby `GrassField` ją rozszerzało to odziedziczyłoby te atrybuty i nie miałoby to dla niej sensu, a wręcz byłoby niespójne modelowo. Generalnie dziedziczenie po nie-abstrakcyjnych klasach wbrew pozorom bardzo rzadko się przydaje w ładnie napisanym programie (w myśl zasady "lepiej stosować kompozycję niż dziedziczenie").